host/frontend/host-orchestrator/operation.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"sync"
+)
+
+type NewOperationError string
+
+func (s NewOperationError) Error() string {
+	return fmt.Sprintf("failure creating a new operation: %s", string(s))
+}
+
+type OperationData struct {
+	Name   string
+	Done   bool
+	Result OperationResultData
+}
+
+type OperationResultData struct {
+	Error OperationErrorData
+	OK    OperationOKData
+}
+
+type OperationErrorData struct {
+	ErrorMsg string
+}
+
+type OperationOKData struct{}
+
+type OperationManager interface {
+	NewOperation() (OperationData, error)
+
+	GetOperation(string) (OperationData, bool)
+
+	CompleteOperation(string, OperationResultData)
+}
+
+type MapOM struct {
+	uuidFactory func() string
```

Comment:
discussed offline and decided to keep the uuid prefix in this case.

Reply:
I thought the conclusion of our offline discussion was to get rid of the factory altogether: The MapOM would always use the uuid library to generate operation names, so it didn't need the factory parameter. The only place where this factory is used without uuid is in a test, where you could simply take the name of the returned operation and assign it to the local variable instead of forcing the value of the local variable to be the name of the operation.

If MapOM receives the factory as a parameter then it doesn't make as much sense to guarantee New would never fail.

Reply:
wanted to keep the parameter for testing purposes to better control the generated operation name to test the panic logic. hid the parameter as suggested and kept as a field to override it in the specific unit test that test the panic logic.